### PR TITLE
ci: change percona-platform to actions

### LINKED ISSUE #1307: Replace percona-platform/setup-go, percona-platform/checkout and percona-platform/cache GH Actions with their upstream variants
### Description

We are using forked GH Actions. Originaly it was made due security reasons, but we figured out that it takes to much effort to support them. So instead lets switch to upstream actions with specified version, example:
```
upsidr/merge-gatekeeper@v1.1.1
```

When PR will be ready we should ask repo admin (@atymchuk) to allow this particular versions in repo settings.

P.s. If you will find another `percona-platform/*` actions in our CI you can replace them as well.

### Suggested solution

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mintbomb27**
This fixes #1307 

Changes the source of percona-platform sourced actions of the following actions:
    * `percona-platform/checkout`
    * `percona-platform/setup-go`
    * `percona-platform/cache`
    * `percona-platform/setup-node`
    * `percona-platform/action-slack-notify`
 to their upstream versions.

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

**Comment by artemgavrilov** [comment]
@atymchuk Please add these actions to allow-list in repo settings
`actions/checkout@v3.0.2`
`actions/setup-go@v2.1.3`
`actions/cache@v2.1.2`
`actions/setup-node@v3.3.0`
`rtCamp/action-slack-notify@v2.2.0`

**Review by BupycHuk — APPROVED** [review]

**Comment by mintbomb27** [comment]
@BupycHuk @artemgavrilov Would be great if you could add the `hacktoberfest-accepted` label. Thank you!

File: .github/workflows/agent.yml [hunk 1]
```diff
@@ -88,7 +88,7 @@ jobs:
           echo "$HOME/gotip/bin" >> $GITHUB_PATH
 
       - name: Enable Go modules cache
-        uses: percona-platform/cache@v2
+        uses: actions/cache@v2.1.2
```
Comment (atymchuk) [inline .github/workflows/agent.yml:91]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/api.yml [hunk 2]
```diff
@@ -37,11 +37,11 @@ jobs:
 
     steps:
       - name: Check out code
-        uses: percona-platform/checkout@v3
+        uses: actions/checkout@v3.0.2
```
Comment (atymchuk) [inline .github/workflows/api.yml:40]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/api.yml [hunk 3]
```diff
@@ -37,11 +37,11 @@ jobs:
...
 
       - name: Set up Go release
         if: env.GO_VERSION != 'tip'
-        uses: percona-platform/setup-go@v2
+        uses: actions/setup-go@v2.1.3
```
Comment (atymchuk) [inline .github/workflows/api.yml:44]:
```suggestion
        uses: actions/setup-go@v2
```

File: .github/workflows/api.yml [hunk 4]
```diff
@@ -55,7 +55,7 @@ jobs:
           echo "$HOME/gotip/bin" >> $GITHUB_PATH
 
       - name: Enable Go modules cache
-        uses: percona-platform/cache@v2
+        uses: actions/cache@v2.1.2
```
Comment (atymchuk) [inline .github/workflows/api.yml:58]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/clean.yml [hunk 5]
```diff
@@ -36,7 +36,7 @@ jobs:
         env:
           # to avoid error due to `go version` accepting -v flag with an argument since 1.15
           GOFLAGS: ""
-        uses: percona-platform/setup-go@v2
+        uses: actions/setup-go@v2.1.3
```
Comment (atymchuk) [inline .github/workflows/clean.yml:39]:
```suggestion
        uses: actions/setup-go@v2
```

File: .github/workflows/clean.yml [hunk 6]
```diff
@@ -52,19 +52,19 @@ jobs:
           echo "GOROOT=$HOME/gotip" >> $GITHUB_ENV
           echo "$HOME/gotip/bin" >> $GITHUB_PATH
       - name: Check out code into the Go module directory
-        uses: percona-platform/checkout@v3
+        uses: actions/checkout@v3.0.2
```
Comment (atymchuk) [inline .github/workflows/clean.yml:55]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/clean.yml [hunk 7]
```diff
@@ -52,19 +52,19 @@ jobs:
...
         with:
           lfs: true
 
       - name: Enable Go modules cache
-        uses: percona-platform/cache@v2
+        uses: actions/cache@v2.1.2
```
Comment (atymchuk) [inline .github/workflows/clean.yml:60]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/clean.yml [hunk 8]
```diff
@@ -52,19 +52,19 @@ jobs:
...
         with:
           path: ~/go/pkg/mod
           key: ${{ matrix.os }}-go-${{ matrix.go.version }}-modules-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ matrix.os }}-go-${{ matrix.go.version }}-modules-
       - name: Enable Go build cache
-        uses: percona-platform/cache@v2
+        uses: actions/cache@v2.1.2
```
Comment (atymchuk) [inline .github/workflows/clean.yml:67]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/common.yml [hunk 9]
```diff
@@ -33,11 +33,11 @@ jobs:
 
     steps:
       - name: Check out code
-        uses: percona-platform/checkout@v3
+        uses: actions/checkout@v3.0.2
```
Comment (atymchuk) [inline .github/workflows/common.yml:36]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/common.yml [hunk 10]
```diff
@@ -33,11 +33,11 @@ jobs:
...
 
       - name: Set up Go release
         if: env.GO_VERSION != 'tip'
-        uses: percona-platform/setup-go@v2
+        uses: actions/setup-go@v2.1.3
```
Comment (atymchuk) [inline .github/workflows/common.yml:40]:
```suggestion
        uses: actions/setup-go@v2
```

File: .github/workflows/common.yml [hunk 11]
```diff
@@ -51,7 +51,7 @@ jobs:
           echo "$HOME/gotip/bin" >> $GITHUB_PATH
 
       - name: Enable Go modules cache
-        uses: percona-platform/cache@v2
+        uses: actions/cache@v2.1.2
```
Comment (atymchuk) [inline .github/workflows/common.yml:54]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/common.yml [hunk 12]
```diff
@@ -167,10 +167,10 @@ jobs:
       SLACK_MESSAGE: "${{ github.event.inputs.repo || github.repository }}:${{ github.event.inputs.branch || github.head_ref }}"
 
     steps:
-      - uses: percona-platform/checkout@v3
-      - uses: percona-platform/setup-node@v3
+      - uses: actions/checkout@v3.0.2
```
Comment (atymchuk) [inline .github/workflows/common.yml:170]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/common.yml [hunk 13]
```diff
@@ -167,10 +167,10 @@ jobs:
...
+      - uses: actions/setup-node@v3.3.0
```
Comment (atymchuk) [inline .github/workflows/common.yml:171]:
```suggestion
      - uses: actions/setup-node@v3
```

File: .github/workflows/common.yml [hunk 14]
```diff
@@ -167,10 +167,10 @@ jobs:
...
       - name: Slack Notification
-        uses: percona-platform/action-slack-notify@v2
+        uses: rtCamp/action-slack-notify@v2.2.0
```
Comment (atymchuk) [inline .github/workflows/common.yml:173]:
```suggestion
        uses: rtCamp/action-slack-notify@v2
```

File: .github/workflows/common.yml [hunk 15]
```diff
@@ -187,6 +187,6 @@ jobs:
       SLACK_MESSAGE: "Workflow failed: ${{ github.event.inputs.repo || github.repository }}:${{ github.event.inputs.branch || github.head_ref }}"
 
     steps:
-      - uses: percona-platform/checkout@v3
+      - uses: actions/checkout@v3.0.2
```
Comment (atymchuk) [inline .github/workflows/common.yml:190]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/common.yml [hunk 16]
```diff
@@ -187,6 +187,6 @@ jobs:
...
       - name: Slack Notification
-        uses: percona-platform/action-slack-notify@v2
+        uses: rtCamp/action-slack-notify@v2.2.0
```
Comment (atymchuk) [inline .github/workflows/common.yml:192]:
```suggestion
        uses: rtCamp/action-slack-notify@v2
```

File: .github/workflows/doc.yml [hunk 17]
```diff
@@ -19,7 +19,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out code
-        uses: percona-platform/checkout@v3
+        uses: actions/checkout@v3.0.2
```
Comment (atymchuk) [inline .github/workflows/doc.yml:22]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/managed.yml [hunk 18]
```diff
@@ -46,11 +46,11 @@ jobs:
 
     steps:
       - name: Check out code
-        uses: percona-platform/checkout@v3
+        uses: actions/checkout@v3.0.2
```
Comment (atymchuk) [inline .github/workflows/managed.yml:49]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/managed.yml [hunk 19]
```diff
@@ -46,11 +46,11 @@ jobs:
...
 
       - name: Enable Go build cache
         if: matrix.use-cache
-        uses: percona-platform/cache@v2
+        uses: actions/cache@v2.1.2
```
Comment (atymchuk) [inline .github/workflows/managed.yml:53]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/managed.yml [hunk 20]
```diff
@@ -60,7 +60,7 @@ jobs:
 
       - name: Enable Go modules cache
         if: matrix.use-cache
-        uses: percona-platform/cache@v2
+        uses: actions/cache@v2.1.2
```
Comment (atymchuk) [inline .github/workflows/managed.yml:63]:
```suggestion
        uses: actions/cache@v2
```

**Review by atymchuk — CHANGES_REQUESTED** [review]

**Comment by mintbomb27** [comment]
@atymchuk Had given specific versions as it was requested so in #1307 . Have replaced the same with the requested versions. Please check again.

**Review by artemgavrilov — APPROVED** [review]

**Pull request merged**